test/core/rules/test_cases/L036.yml

```diff
@@ -26,7 +37,26 @@ test_multiple_select_targets_all_on_the_same_line:
   fail_str: |
     select a, b, c
     from x
+  # In a normal "sqlfluff fix" run, some formatting issues left over by this
+  # rule would be addressed by other rules.
+  # TRICKY: Use explicit newlines to preserve the trailing space after "SELECT".
+  fix_str: "select \na, b, c\nfrom x\n"
 
 test_multiple_select_targets_trailing_whitespace_after_select:
   # TRICKY: Use explicit newlines to preserve the trailing space after "SELECT".
   pass_str: "SELECT \n    a,\n    b\nFROM t\n"
+
+test_comment_between_select_and_single_select_target:
+  fail_str: |
+    SELECT
+        -- This is the user's ID.
+         user_id
+    FROM
+        safe_user
+  # In a normal "sqlfluff fix" run, some formatting issues left over by this
+  # rule would be addressed by other rules.
+  fix_str: |
+    SELECT user_id
+        -- This is the user's ID.
```

Comment:
Reviewers: Note that after the fix, the comment now appears misplaced. I'm thinking the rule should try and address this. What do you think? Are you aware of other rules doing similar things that I could borrow from?

Side note: This reminds me a bit of ["widow and orphan" contro](https://support.microsoft.com/en-us/office/control-pagination-e9b3b005-cf62-41d0-afa2-24cfb223ab42)l in word processors, where the word processor tries to avoid leaving a line or table row by itself on a page.